packages/hawtio/src/plugins/camel/debug/debug-service.ts

```diff
@@ -0,0 +1,301 @@
+import { eventService } from '@hawtiosrc/core'
+import { MBeanNode } from '@hawtiosrc/plugins/shared'
+import { log } from '../globals'
+import * as ccs from '../camel-content-service'
+import { jolokiaService } from '@hawtiosrc/plugins/connect'
+import { camelPreferencesService } from '../camel-preferences-service'
+import { IRequest, IResponseFn } from 'jolokia.js'
+
+export interface ConditionalBreakpoint {
+  nodeId: string
+  language: string
+  predicate: string
+}
+
+export interface MessageData {
+  id: string | null
+  uid: string
+  timestamp: string
+  headers: Record<string, string>
+  headerTypes: Record<string, string>
+  headerHtml: string
+  body?: string
+  bodyType?: string
+  toNode?: string
+}
+
+class DebugService {
+  private handles: number[] = []
+
+  notifyError(msg: string) {
+    eventService.notify({
+      type: 'danger',
+      message: msg,
+    })
+  }
+
+  notifyInfo(msg: string) {
+    eventService.notify({
+      type: 'info',
+      message: msg,
+    })
+  }
+
+  async register(request: IRequest, callback: IResponseFn) {
+    const handle = await jolokiaService.register(request, callback)
+    log.debug('Register handle:', handle)
+    this.handles.push(handle)
+  }
+
+  unregisterAll() {
+    log.debug('Unregister all handles:', this.handles)
+    this.handles.forEach(handle => jolokiaService.unregister(handle))
+    this.handles = []
+  }
+
+  getDebugBean(node: MBeanNode): MBeanNode | null {
+    const db = ccs.findDebugBean(node) as MBeanNode
+    if (!db || !db.objectName) this.notifyError('Could not find the debug bean')
+
+    return db
+  }
+
+  async isDebugging(node: MBeanNode): Promise<boolean> {
+    const db = this.getDebugBean(node)
+    if (!db) return false
+
+    const result = await jolokiaService.readAttribute(db.objectName as string, 'Enabled')
+    if (!result) return false
+
+    return result as boolean
+  }
+
+  async setDebugging(node: MBeanNode, flag: boolean): Promise<boolean> {
+    const db = this.getDebugBean(node)
+    if (!db) return false
+
+    const max = flag ? 1 : 0
+    camelPreferencesService.saveCamelPreferences({ maximumTraceDebugBodyLength: max })
+
+    const streams = camelPreferencesService.loadCamelPreferences().isIncludeTraceDebugStreams
+
+    await jolokiaService.writeAttribute(db.objectName as string, 'BodyMaxChars', max)
```

Comment:
why do we need `as string`?  if it's just because of converting `string | undefined` to `string`, rather we should check the existence of `objectName` before the sequence of the method calls to make type check satisfied.